File: elixir/raindrops/raindrops.exs
defmodule Raindrops do
  @doc """
  Returns a string based on raindrop factors.

  - If the number contains 3 as a prime factor, output 'Pling'.
  - If the number contains 5 as a prime factor, output 'Plang'.
  - If the number contains 7 as a prime factor, output 'Plong'.
  - If the number does not contain 3, 5, or 7 as a prime factor,
    just pass the number's digits straight through.
  """
  @spec convert(pos_integer) :: String.t
  def convert(n) do
    string = Enum.join([
      (if rem(n, 3) == 0, do: "Pling"),
      (if rem(n, 5) == 0, do: "Plang"),
      (if rem(n, 7) == 0, do: "Plong"),
    ])

    if string == "", do: to_string(n), else: string
  end
end

File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(num)
    output = [
      ("Pling" if num % 3 == 0),
      ("Plang" if num % 5 == 0),
      ("Plong" if num % 7 == 0)
    ].join

    output.empty? ? num.to_s : output
  end
end

module BookKeeping; VERSION = 2; end
